src/lib/markbind/src/parser.js

```diff
@@ -685,15 +429,71 @@ class Parser {
     return new Promise((resolve, reject) => {
       let actualFilePath = file;
       if (!utils.fileExists(file)) {
-        const boilerplateFilePath = Parser.calculateBoilerplateFilePath(path.basename(file), file, config);
+        const boilerplateFilePath = urlUtils.calculateBoilerplateFilePath(path.basename(file), file, config);
         if (utils.fileExists(boilerplateFilePath)) {
           actualFilePath = boilerplateFilePath;
         }
       }
 
-      this.preprocess(actualFilePath, pageData, context, config)
-        .then(resolve)
-        .catch(reject);
+      const currentContext = context;
```

Comment:
This chunk is moved from the `preprocess` ( not `_preprocess` ) function, which was added in the expressive layouts pr,
to remove naming ambiguity.

A future PR may look into whether expressive layouts can be introduced in similar fashion to the page `header`, `sitenav`, etc., as `includeData` ( the function this code is under ) is very similar to `includeFile`. ( to reduce code duplication )